src/services/FeedProcessor/CoingeckoMultiProcessor.ts

```diff
@@ -1,18 +1,50 @@
 import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
 
 import {InputParams, OutputValues} from '../fetchers/CoingeckoPriceMultiFetcher.js';
-
-import {FeedFetcher} from '../../types/Feed.js';
+import {PriceDataRepository, PriceDataPayload} from '../../repositories/PriceDataRepository.js';
 import {CoingeckoPriceMultiFetcher} from '../fetchers/index.js';
+import {FeedFetcher} from '../../types/Feed.js';
+import {FetcherName} from '../../types/fetchers.js';
+import TimeService from '../TimeService.js';
+import {feedNameToBaseAndQuote} from '../../utils/hashFeedName.js';
 
 @injectable()
 export default class CoingeckoMultiProcessor {
   @inject(CoingeckoPriceMultiFetcher) coingeckoPriceMultiFetcher!: CoingeckoPriceMultiFetcher;
+  @inject(PriceDataRepository) private priceDataRepository!: PriceDataRepository;
+  @inject(TimeService) private timeService!: TimeService;
+  @inject('Logger') private logger!: Logger;
+
+  static fetcherSource = '';
 
   async apply(feedFetchers: FeedFetcher[]): Promise<(number | undefined)[]> {
     const params = this.createParams(feedFetchers);
     const outputs = await this.coingeckoPriceMultiFetcher.apply(params);
 
+    const payloads: PriceDataPayload[] = [];
+    for (const [ix, output] of outputs.entries()) {
+      try {
+        const [feedBase, feedQuote] = feedNameToBaseAndQuote(feedFetchers[ix].symbol || '-');
+
+        if (output) {
+          payloads.push({
+            fetcher: FetcherName.COINGECKO_PRICE,
+            value: output.value.toString(),
+            valueType: 'string',
```

Comment:
let's create `enum` for value type, and all current fetcher will have type `price`.

Reply:
done